Make reply a thenable (#1869)
fastify/fastify: docs/Reply.md

```

If you want to know more please review [Routes#async-await](https://github.com/fastify/fastify/blob/master/docs/Routes.md#async-await).

<a name="then"></a>
### .then(fullfilled, rejected)

> As the name suggests, `reply.then()` makes `Reply` awaitable.

Reviewer: I'd refer to some source which explains "await able". We can also use it with promises only.

```

If you want to know more please review [Routes#async-await](https://github.com/fastify/fastify/blob/master/docs/Routes.md#async-await).

<a name="then"></a>
### .then(fullfilled, rejected)

> As the name suggests, a `Reply` object can be awaited up, i.e. `await reply` will wait until the reply is sent.

Reviewer 2: ```suggestion
As the name suggests, a `Reply` object can be awaited upon, i.e. `await reply` will wait until the reply is sent.
```

The `await` syntax calls the `reply.then()`.

`reply.then(fullfilled, rejected)` accepts two parameters:

- `fullfilled` will be called when a response has been fully sent,

> - `rejected` will be called if the underlining stream had an error, e.g.

Reviewer 2: ```suggestion
- `rejected` will be called if the underlying stream had an error, e.g.
```

the socket has been destroyed.

For more details, see:

> - https://github.com/fastify/fastify/issues/1864 for the discussions

Reviewer 2: ```suggestion
- https://github.com/fastify/fastify/issues/1864 for the discussion about this feature
```

> - https://promisesaplus.com/ for the definition of thenable

Reviewer 2: ```suggestion
- https://promisesaplus.com/ for the definition of thenables
```

> - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then for the signature

Reviewer 2: The link right above this one is to the spec which covers the signature. Is this one really necessary?

Reviewer 2: 👍
